skimage/filters/tests/test_ridges.py

```diff
@@ -14,8 +14,8 @@ def test_2d_null_matrix():
     zeros = np.zeros((3, 3))
     ones = np.ones((3, 3))
 
-    assert_equal(meijering(a_black, black_ridges=True), zeros)
-    assert_equal(meijering(a_white, black_ridges=False), zeros)
+    assert_equal(meijering(a_black, black_ridges=True), ones)
```

Comment:
Isn't this a big behavioral change? Was the old one always blatantly wrong?

Reply:
The previously implemented formulas were wrong, yes, unfortunately. The tests in question check edge cases, namely how the filter deals with all-dark/bright images. The relevant part is that the filtered image is roughly constant again and no structures are created 'out of nothing' (apart from border effects perhaps). That the result was '0' before and is '1' now is related to the different sorting of eigenvalues, but I can't say more about it..